keps/sig-cli/2379-kubectl-plugins/README.md

```diff
@@ -191,6 +192,46 @@ kubectl myplugin __complete "$@"
 ```
 Then all that is needed is to name this file `kubectl_complete-myplugin`, put it somewhere on `$PATH` and give it executable permissions.
 
+#### Built-in Command Shadowing Support
+
+There is a heavy demand for enriching some built-in kubectl commands with new subcommands. More specifically take an instance of `create`
+command, even though it supports the creation of 18 resources(as of v1.25) with one line command execution(e.g. `kubectl create job`), 
+new requests to support other resources as well are occasionally filed. These requests are understandable
+from the user's point of view, on the contrary, it is not sustainable in terms of the code base. Because that might bring about not only increase
+the size of the binary but also create a maintenance burden.
+
+As an extension of this proposal, this section proposes a viable alternative to users via plugins instead of creating static subcommands
+inside kubectl built-in commands.
+
+This extension introduces a new boolean flag, namely `--use-plugin`. Without this flag, plugin mechanism will work
```

Comment:
For alpha, I'd suggest using an environment variable `KUBECTL_PLUGIN_SHADOW_CREATE` or something like that, which is consistent with other mechanisms we've been implementing over the years.

Reply:
Thanks. I proposed `KUBECTL_ENABLE_ALPHA_CMD_SHADOW`, I'm open to change that.